stumpy/stomp.py

```diff
@@ -101,36 +117,49 @@ def stomp(T_A, m, T_B=None, ignore_trivial=True):
         IR = -1  # No left and right matrix profile available
     out[0] = P, I, -1, IR
 
+    T_B[np.isnan(T_B)] = 0
+
+    QT = core.sliding_dot_product(T_B[:m], T_A)
+    QT_first = core.sliding_dot_product(T_A[:m], T_B)
+
     k = T_A.shape[0] - m + 1
     for i in range(1, l):
         QT[1:] = (
             QT[: k - 1] - T_B[i - 1] * T_A[: k - 1] + T_B[i - 1 + m] * T_A[-(k - 1) :]
         )
         QT[0] = QT_first[i]
-        D = core.calculate_distance_profile(
-            m, QT, μ_Q[i].item(0), σ_Q[i].item(0), M_T, Σ_T
-        )
+
+        D = core.calculate_distance_profile(m, QT, μ_Q[i], σ_Q[i], M_T, Σ_T)
```

Comment:
Would you mind adding the `.item(0)` back as this helps to ensure that the type being passed over is a scalar and not an array?